packages/marshal/src/marshal.js

```diff
@@ -984,7 +1018,7 @@ export function makeMarshal(
             );
             const result = ibidTable.start({});
             result[QCLASS] = fullRevive(rawTree.original);
-            if ('rest' in rawTree) {
+            if (rawTree.rest !== undefined) {
```

Comment:
This weakens the actual test. How can we keep the narrower meaning but still have it play well with the typing?

Reply:
I'm not sure.  Do note that the `rest` property is guaranteed not to be `undefined` in a proper encoding.  Would you rather explicitly reject the appearance of a `rest` that is undefined?

Reply:
Yes, explicitly reject. Would that also be easy to type?